observability/probe/aggregate.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package probe
+
+import (
+	"github.com/juju/errors"
+)
+
+type Aggregate struct {
+	// Probes is a map of probes to run as part of this aggregate with the key
+	// corresponding to well known name for the probe.
+	Probes map[string]Prober
```

Comment:
if the keys are well known names we should use a type alias

Reply:
They can be anything. Just a canonical name to reference the supplier of the probe. As discussed.